src/formatter.py

```diff
@@ -0,0 +1,125 @@
+
+
+class DictFormatter:
+    '''
+    Provides methods of serialisation of Game instance for various viewpoints
+    '''
+
+    def __init__(self, game):
+        self.game = game
+
+    def to_dict(self):
+        return DictFormatter.game_to_dict(self.game)
+
+    @classmethod
+    def game_to_dict(cls, game):
+        """
+        :param game: Game object
+        :return: dict with (almost) complete state of the game. Decks return only the discard.
+        :players: dict of (id => Player dicts)
+            :name: str
+            :location: str
+            :action_count: int
+            :hand: dict of (id => Card dicts)
+                :name: str
+                :colour: str
+        :player_deck_discard: dict of (id => Card dicts)
+            :name: str
+            :colour: str
+        :infect_deck_discard: dict of (id => Card dicts)
+            :name: str
+            :colour: str
+        :diseases: dict of (colour => Disease dicts)
+            :colour: str
+            :cured: Boolean
+            :public_health: int
+        :cities: dict of (name => City dicts)
+            :name: str
+            :has_lab: Boolean
+            :colour: str
+            :infection_levels: dict (colour => lvl)
+        :infection_rate: int
+        :epidemic_count: int
+        """
+        output = {}
+        output['players'] = {id: cls.player_to_dict(player) for id, player in enumerate(game.players)}
+        output['player_deck_discard'] = cls.deck_to_dict(game.player_deck)
+        output['infect_deck_discard'] = cls.deck_to_dict(game.infect_deck)
+        output['diseases'] = {colour: cls.disease_to_dict(disease) for colour, disease in
+                              game.diseases.items()}
+        output['cities'] = {name: cls.city_to_dict(city) for name, city in game.city_map.items()}
+        output['infection_rate'] = game.infection_rate
+        output['epidemic_count'] = game.epidemic_count
+
+        return output
+
+    @classmethod
+    def card_to_dict(cls, card):
+        """
+        :param card: Card object
+        :return: New dict with Card properties
+        :name: str
+        :colour: str
+        """
+        return {'name': card.name, 'colour': card.colour}
+
+    @classmethod
+    def deck_to_dict(cls, deck):
+        """
+        :param deck: Deck object
+        Does not return still uncovered cards
+        :return: dict (id => card dict), card objects are copied
+        """
+        return {id: DictFormatter.card_to_dict(card) for id, card in enumerate(deck.discard)}
```

Comment:
Do we use need to use `DictFormatter` as `cls` here? If so, it's better to change it.